pkg/encoding/codegen/fields.go

```diff
@@ -915,9 +915,14 @@ func (f *FixedUintField) GenReadFrom() (string, error) {
 		g.printlnf("{")
 		g.printlnf("tempVal := %s(0)", digit)
 		gen("tempVal")
-		g.printlnf("value." + f.name + " = &tempVal")
+		g.printlnf("value.%s = &tempVal", f.name)
 		g.printlnf("}")
-		return g.output()
+
+		// Special case for a single byte - directly use wire address
+		// This is useful to modify the TLV in-place (e.g. HopLimit)
+		if f.l == 1 {
+			g.printlnf("value.%s = &reader.Range(reader.Pos()-1, reader.Pos())[0][0]", f.name)
```

Comment:
In this case, I think you'd better remove the code generated above in `{...}` since they are useless.

Reply:
But then I need to check for eof here manually. This seemed shorter?

Reply:
I think you can just write `if f.l != 1 { ... } else { ... }`

Reply:
I meant - if I understand right, `reader.Range(pos, pos+1)` will panic if there's EOF after `pos`. So I need to move ahead first, check for EOF, then do range - no?

Reply:
You can use `pos+1` I think.

Reply:
Switched to `Skip()` which should be cleaner. I wrote a small test for this and using `Range(pos, pos+1)` does panic without a error check.

```go
package main

import (
	"fmt"

	"github.com/zjkmxy/go-ndn/pkg/encoding"
	spec "github.com/zjkmxy/go-ndn/pkg/ndn/spec_2022"
	"github.com/zjkmxy/go-ndn/pkg/utils"
)

func main() {
	name, _ := encoding.NameFromStr("/hello/world")

	i := &spec.Interest{
		NameV:     name,
		HopLimitV: utils.IdPtr(uint8(7)),
	}

	encoder := spec.InterestEncoder{}
	encoder.Init(i)
	wire := encoder.Encode(i).Join()

	if wire[len(wire)-1] != byte(7) {
		fmt.Println("last byte is not 7")
		return
	}

	reader := encoding.NewBufferReader(wire)
	context := spec.InterestParsingContext{}
	context.Init()
	i_p, err := context.Parse(reader, false)
	if err != nil {
		fmt.Println(err)
		return
	}

	if *i_p.HopLimitV != 7 {
		fmt.Println("hop limit is not 7")
		return
	}

	*i_p.HopLimitV = byte(9)
	fmt.Println(wire)
	if wire[len(wire)-1] != byte(9) {
		fmt.Println("last byte is not 9")
		return
	}

	// Attempt to parse with one byte less
	reader = encoding.NewBufferReader(wire[:len(wire)-1])
	context = spec.InterestParsingContext{}
	context.Init()
	_, err = context.Parse(reader, false)
	if err == nil {
		fmt.Println("should have failed")
		return
	}

	fmt.Println("success")
}
```

Without the error check for `Skip()` and `reader.Range(reader.Pos(), reader.Pos()+1)`

```
root@0037b98ec2ac:/work/go-ndn# go run tt.go 
[7 14 8 5 104 101 108 108 111 8 5 119 111 114 108 100 34 1 9]
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/zjkmxy/go-ndn/pkg/ndn/spec_2022.(*InterestParsingContext).Parse(0xc0001e9de0, {0x5ffd48, 0xc00017c2a0}, 0x0)
        /work/go-ndn/pkg/ndn/spec_2022/zz_generated.go:3939 +0xfc5
main.main()
        /work/go-ndn/tt.go:53 +0x39c
exit status 2
```

with the EOF check and `reader.Range(reader.Pos()-1, reader.Pos())`

```
root@0037b98ec2ac:/work/go-ndn# go run tt.go 
[7 14 8 5 104 101 108 108 111 8 5 119 111 114 108 100 34 1 9]
success
```

Reply:
Sorry I forgot `Range` only works for already read parts